Lib/test/test_csv.py

```diff
@@ -1018,6 +1019,16 @@ def test_doublequote(self):
         dialect = sniffer.sniff(self.sample9)
         self.assertTrue(dialect.doublequote)
 
+    def test_guess_lineterminator(self):
+        sniffer = csv.Sniffer()
+        dialect = sniffer.sniff(self.sample5)
+        self.assertEqual(dialect.lineterminator, '\r\n')
+        dialect = sniffer.sniff(self.sample8)
+        self.assertEqual(dialect.lineterminator, '\n')
+        dialect = sniffer.sniff(self.sample10)
+        self.assertEqual(dialect.lineterminator, '\r')
+
```

Comment:
Add tests to cover the last line of _guess_lineterminator: `return os.linesep`.  Add a test with the same literal, but with an oddball terminator, such as '\v' (vertical tab), and another with a single line with no control chars.